wpilibj/src/main/java/edu/wpi/first/wpilibj/DriverStation.java

```diff
@@ -1025,22 +1025,39 @@ private void sendMatchData() {
     int matchNumber;
     int replayNumber;
     int matchType;
+    int controlWord;
     synchronized (m_cacheDataMutex) {
       eventName = m_matchInfo.eventName;
       gameSpecificMessage = m_matchInfo.gameSpecificMessage;
       matchNumber = m_matchInfo.matchNumber;
       replayNumber = m_matchInfo.replayNumber;
       matchType = m_matchInfo.matchType;
+      controlWord = HAL.nativeGetControlWord();
```

Comment:
Is there any reason why this has been moved into this critical section?

Reply:
It needed to be cached. Should the assignment be outside the `synchronized ` block?

Reply:
Yes, it can be outside the synchronized block (the synchronized block is protecting the contents of m_matchInfo.

However, the approach you are using here is not what we want to do, as calling get() is relatively expensive.  Instead you should cache the actual values.  E.g. m_matchDataSender should have a copy of the last set values.

Reply:
Is there a reason to not refactor this into a method in `MatchDataSender`?

Reply:
It can be refactored, sure.